File: java-spark-demo/src/main/java/SimpleApp.java
import org.apache.spark.api.java.*;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.function.Function;

public class SimpleApp {
    public static void main(String[] args) {
        String logFile = "file:///Users/css/data/test.txt"; // Should be some file on your system
        SparkConf conf = new SparkConf().setAppName("Simple Application").setMaster("local");

        JavaSparkContext sc = new JavaSparkContext(conf);
        JavaRDD<String> logData = sc.textFile(logFile).cache();

        long numHello = logData.filter(new Function<String, Boolean>() {
            public Boolean call(String s) {
                return s.contains("hello");
            }
        }).count();

        long numWorld = logData.filter(new Function<String, Boolean>() {
            public Boolean call(String s) {
                return s.contains("world");
            }
        }).count();

        System.out.println("Lines with hello: " + numHello + ", lines with world: " + numWorld);

        sc.stop();
    }
}

File: scala-spark-app/src/main/scala/com/lz/demo/SimpleApp.scala
package com.lz.demo

import org.apache.spark.{SparkConf, SparkContext}

object SimpleApp {
    def main(args: Array[String]): Unit = {
        val logFile = "input.txt"
        val conf = new SparkConf().setAppName("Simple Application").setMaster("local")
        val sc = new SparkContext(conf)
        val logData = sc.textFile(logFile, 2).cache()
        val numAs = logData.filter(line => line.contains("a")).count()
        val numBs = logData.filter(line => line.contains("b")).count()
        println("Lines with a: %s, Lines with b: %s".format(numAs, numBs))
    }
}
